experimental/modular_oak_functions/src/io.rs

```diff
@@ -0,0 +1,42 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::demux::Demux;
+
+pub struct IoListener {
+    demux: Demux,
+}
+
+impl IoListener {
+    pub fn new(demux: Demux) -> Self {
```

Comment:
Will it take a socket file as input?

Reply:
Updated the comment above. In different environments we will have different implementation of the IoListener: e.g. vsock or UDS listener for Linux-based implementations running in a VM, serial listener (or virtio-pci) for the UEFI implementation, etc.